modules/mobile/app/helpers/mobile/list_filter.rb

```diff
@@ -2,54 +2,131 @@
 
 module Mobile
   class ListFilter
-    def initialize(collection, filters)
+    include SentryLogging
+
+    class FilterError < StandardError
+    end
+
+    PERMITTED_OPERATIONS = %w[eq notEq].freeze
+
+    def initialize(collection, filter_params)
       @collection = collection
-      @filters = filters
+      @filter_params = filter_params
     end
 
     # Accepts params:
     #   @collection - a Common::Collection of Common::Base models
-    #   @filters - should be an ActionController::Parameters object which should be passed in from the
+    #   @filter_params - should be an ActionController::Parameters object which should be passed in from the
     #     controller via @params[:filter]. This will pass in another ActionController::Parameters object.
     # Returns: a new Common::Collection of Common::Base models that match the provided filters
-    def self.matches(collection, filters)
-      filterer = new(collection, filters)
+    def self.matches(collection, filter_params)
+      filterer = new(collection, filter_params)
       filterer.result
     end
 
-    # carrying over errors and metadata while merging filters into metadata because
-    # that's how the Common::Collection filter works. these values are probably not used again
     def result
```

Comment:
It wasn't clear what should happen when we encounter an error. Because this is very new and may have some kinks to be worked out, I've chosen to log the error to sentry, add the error to the collection (which will later add it to the serializer metadata), and return the full collection. Alternatively, we could just let it error and return 422 to the mobile app, but I chose the lighter touch approach.